client/src/actions/auth/passwordAction.js

```diff
@@ -0,0 +1,50 @@
+import axios from 'axios';
+import { browserHistory } from 'react-router';
+import actionTypes from '../actionTypes';
+
+export const recoverPasswordAction = userEmail => (dispatch) => {
+  axios.post('/api/user/recover-password', userEmail)
+    .then((res) => {
+      if (res.status !== 200) {
+        const payload = 'Something went wrong...Try again';
+        return dispatch({ type: actionTypes.RECOVERY_UNSUCCESSFUL, payload });
+      }
+      const payload = res.data; // password link sent
+      return dispatch({ type: actionTypes.RECOVERY_SUCCESSFUL, payload });
+    })
+    .catch((err) => {
+      if (err && err.response.data.validateError) {
```

Comment:
Awesome stuff! I like the way you account for different errors possible. Have you considered using a switch statement instead of the if statements?